app/models/manageiq/providers/kubernetes/container_manager/refresh_parser.rb

```diff
@@ -362,8 +362,28 @@ def parse_container(container, pod_id)
         :backing_ref     => container.containerID,
         :container_image => parse_container_image(container.image, container.imageID)
       }
+      state_attributes = parse_container_state container.lastState
+      state_attributes.each { |key, val| h[key.to_s.prepend('last_').to_sym] = val } if state_attributes
+      h.merge! parse_container_state container.state
+    end
+
+    def parse_container_state(state_hash)
```

Comment:
Maybe have this take an additional `prefix` argument:

```ruby
   def parse_container_state(state_hash, prefix = "")
     ...
     res["#{prefix}state".to_sym] = state
     %w(reason started_at finished_at exit_code signal message).each do |attr|
       res["#{prefix}#{attr}".to_sym] = state_info[attr.camelize(:lower)]
     end
     ...
   end
```

Then the caller could just do:

```ruby
  h.merge! parse_container_state(container.lastState, "last_")
  h.merge! parse_container_state(container.state)
```

The way it reads currently, there's a bit of extra parsing that gets in the middle of understanding what's really being parsed.

The key here is to make the parsing as clear as possible in as many places as possible.

Thoughts?